File: java/com/aa/act/interview/org/Employee.java
package com.aa.act.interview.org;

public class Employee {

	private int identifier;
	private  Name name;

	public Employee(int identifier, Name name) {
		if(name == null)
			throw new IllegalArgumentException("name cannot be null");
		this.identifier = identifier;
		this.name = name;
	}
// added empty constructor
	public Employee() {
	}

// added a new one param constructor for name
	public Employee(Name person) {
		this.name = person;
	}
	
	public int getIdentifier() {
		return identifier;
	}
	
	public Name getName() {
		return name;
	}

	@Override
	public String toString() {
		return name.toString() + ": " + identifier;
	}
}

File: dotnet/MyOrganization/Employee.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace MyOrganization
{
    internal class Employee
    {
        private int identifier;
        private Name name;

        public Employee(Name name, int Identifier)
        {
            if (name == null)
                throw new Exception("name cannot be null");
            this.identifier = Identifier;
            this.name = name;
        }

        public int GetIdentifier()
        {
            return identifier;
        }

        public Name GetName()
        {
            return name;
        }

        override public string ToString()
        {
            return name.ToString();
        }
    }
}
